java/java.source.base/src/org/netbeans/modules/java/source/parsing/PatchModuleFileManager.java

```diff
@@ -296,7 +296,8 @@ public FileObject getFileForOutput(Location location, String packageName, String
 
     private Location fixLocation(Location input) throws IOException {
         if (input == StandardLocation.CLASS_OUTPUT && overrideModuleName != null) {
-            return this.moduleLocations.stream().filter(pl -> overrideModuleName.equals(pl.getModuleName())).map(pl -> (Location) pl).findAny().orElse(input);
+            return moduleLocations(StandardLocation.PATCH_MODULE_PATH)
+                    .stream().filter(pl -> overrideModuleName.equals(pl.getModuleName())).map(pl -> (Location) pl).findAny().orElse(input);
```

Comment:
this looks good to me, I actually wanted to propose the same but you were faster this time :)

The confusing part while looking through the code however was that `moduleLocations(foo)` only allows foo to be the constant `StandardLocation.PATCH_MODULE_PATH`, everything else throws an exception. So it would be more intuitive to simply make that a `getModueLocations()` without parameter.

Reply:
Got to be faster than you sometime! :smile: I kept the fix minimal but had similar thoughts wondering why that method takes the location.  Mind you `moduleLocations()` would be fine - no need for `get` on everything!